test cases/frameworks/6 gettext/data/meson.build

```diff
@@ -27,4 +27,31 @@ i18n.merge_file(
   install_dir: join_paths(get_option('datadir'), 'applications')
 )
 
+# Regression test: https://github.com/mesonbuild/meson/issues/9022
```

Comment:
Could you add specifically "passing a File object as input", so that the next person doesn't have to open github to figure this out?

Reply:
ok, done.